### ISSUE TITLE
Add the function "postpone" for event handlers.

### ISSUE THREAD

**Issue by gallandarakhneorg** [sarl/sarl#288]
Consider the following SARL definitions:

```
agent A1 {
   on Initialize {
      doSometing
      emit(new AgentInitialized)
   }
}
agent A2 extends A1 {
   on Initialize {
      super._handler_Initialize_0(occurrence)
      doSometingLong
   }
}
```

But the `AgentInitialized` is not fired at the correct instant when the initialized agent is `A2`: doSomethingLong is run after the event was fired.

I propose to add a function in the `Schedules` capacity for differing the execution of a code at the end of the current event handler. The code will becomes:

```
agent A1 {
   on Initialize {
      doSometing
      postpone [ emit(new AgentInitialized) ]
   }
}
agent A2 extends A1 {
   on Initialize {
      super._handler_Initialize_0(occurrence)
      doSometingLong
   }
}
```

The new code ensures that the `AgentInitialized` event is fired after all the event handling functions are run in the inheritance hierarchy.

**Comment by gallandarakhneorg** [comment]
All the event handlers must be run in parallel, even inside super types.

**Comment by srodriguez** [comment]
So this code should provide the equivalent for:

´´´
agent A3 {
   on Initialize {
      doSometing
      differ [ emit(new AgentInitialized) ]
   }
   on Initialize {

      doSometingLong
   }
}
´´´
Still to discuss about differ

**Comment by srodriguez** [comment]
The semantics of agent's extend will be discussed in #339

**Comment by gallandarakhneorg** [comment]
I propose to change the name from "differ" to "postpone".

**Comment by gallandarakhneorg** [comment]
According to #339, the initial problem may be rewritten:
```
agent A1 {
   def initialization {
      doSometing
      emit(new AgentInitialized)
   }
   on Initialize {
        initialization()
   }
}
agent A2 extends A1 {
   def initialization {
      super.initialization()
      doSometingLong
   }
}
```

If we assumed that the events fired during the "Initialize" execution as buffered and emitted after the all the initialize handlers are ran (@ngaud has implemented this in Janus, I am right?). The postpone function is no more needed.

Nevertheless, should we consider to add the postpone function for events that are neither Initialize or Destroy (they are already concerned by the event buffering mechanism)?
And/or should we include in the syntax a flag that enables to turn on to event buffering mechanism for event handlers?

If the first question is true, this issue is still valid.
If the second question is true, we need to open a new issue.

**Comment by gallandarakhneorg** [comment]
Add document in the "SARL best practises".
